Extensions/ArtifactEngine/IntegrationTests/proxyTests.ts

```diff
@@ -0,0 +1,119 @@
+                res.write(JSON.stringify({
+                    "artifacts": [
+                        { "displayPath": "file1.pdb", "fileName": "file1.pdb", "relativePath": "Extensions/ArtifactEngine/TestData/Jenkins/file1.pdb" },
+                    ]
+                }));
+                res.end();
+                return;
+            }
+
+            if (req.url === "/job/ArtifactEngineJob/6/artifact/Extensions/ArtifactEngine/TestData/Jenkins/file1.pdb") {
+                res.writeHead(200, { 'Content-Type': 'text/plain' });
+                res.write('dummyFileContent');
+                res.end();
+                return;
+            }
+
+            res.writeHead(404, { 'Content-Type': 'text/plain' });
+            res.write('page not found!');
+            res.end();
+        });
+
+        proxy.on('connect', onConnect);
+        function onConnect(req, clientSocket, head) {
+            assert.equal(req.headers['proxy-authorization'], 'Basic ' + new Buffer('admin:123:pass#123:').toString('base64'));
+
+            var serverSocket = net.connect({ port: 9011 }, function () {
+                clientSocket.write('HTTP/1.1 200 Connection established\r\n\r\n');
+                clientSocket.pipe(serverSocket);
+                serverSocket.write(head);
+                serverSocket.pipe(clientSocket);
+                // workaround, see joyent/node#2524
+                serverSocket.on('end', function () {
+                    clientSocket.end();
+                });
+            });
+        };
+
+        proxy.listen(9011, setUpClient);
+
+        function setUpClient() {
+            let processor = new engine.ArtifactEngine();
+            let processorOptions = getArtifactEngineOptions();
+            let webProvider = getJenkinsWebProvider();
+            let stubProvider = new providers.StubProvider();
+
+            var processItemsPromise = processor.processItems(webProvider, stubProvider, processorOptions);
+            processItemsPromise.then((tickets) => {
+                assert.equal(stubProvider.itemsUploaded["Extensions/ArtifactEngine/TestData/Jenkins/file1.pdb"], "dummyFileContent");
+                assert.equal(tickets.find(x => x.artifactItem.path == "Extensions/ArtifactEngine/TestData/Jenkins/file1.pdb").retryCount, 0);
+                proxy.close();
+                done();
+            }, (err) => {
+                throw err;
+            });
+        }
+    });
+});
+
+function getArtifactEngineOptions(): engine.ArtifactEngineOptions {
+    let processorOptions = new engine.ArtifactEngineOptions();
+    processorOptions.fileProcessingTimeoutInMinutes = 5;
+    processorOptions.itemPattern = "**";
+    processorOptions.parallelProcessingLimit = 8;
+    processorOptions.retryIntervalInSeconds = 0;
+    processorOptions.retryLimit = 2;
+    processorOptions.verbose = true;
+
+    return processorOptions;
+}
+
+function getJenkinsWebProvider(): providers.WebProvider {
```

Comment:
this test doesn't contact Jenkins server, right?

Reply:
@GitHubSriramB no, it doesn't.